packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/model/clientmodel/EnumType.java

```diff
@@ -190,7 +205,7 @@ public String jsonSerializationMethodCall(String jsonWriterName, String fieldNam
             ? valueGetter + "." + getToMethodName() + "()"
             : valueGetter + " == null ? null : " + valueGetter + "." + getToMethodName() + "()";
 
-        return elementType.asNullable().jsonSerializationMethodCall(jsonWriterName, fieldName, actualValueGetter,
+        return wireType.asNullable().jsonSerializationMethodCall(jsonWriterName, fieldName, actualValueGetter,
             jsonMergePatch);
```

Comment:
What's the difference of elementType with wireType?

Is elementType the client type of wireType?

Reply:
This is specially for Swagger non-string expandable enum.

Currently we always map expandable enum to `ExpandableStringEnum` in Swagger. Current Swagger is having trouble doing stream-style serialization, as it uses elementType to write serialization.
e.g. if elementType is int, then in `toJson` method, this field is serialized as 
```java
jsonWriter.writeNumberField("priorityModel", this.priorityModel == null ? null : this.priorityModel.toInt());
```
which should be
```java
jsonWriter.writeStringField("priorityModel", this.priorityModel == null ? null : this.priorityModel.getValue()); // or this.priorityModel.toString()
```

Reply:
I see. "getValue" and "toString" is the same thing. Sure, make sense.

Reply:
Yes, for `ExpandableStringEnum`, currently I explicitly set the wireType to `String` to match the workaround I did for `monitor` and `cdn`(depends on what `modelAsString` really means in Swagger).